hooks/tk-multi-publish2/basic/publish_session.py

```diff
@@ -0,0 +1,712 @@
+        <b><a href='%s'>Loader</a></b> so long as they have access to the file's
+        location on disk.
+
+        If the session has not been saved, validation will fail and a button
+        will be provided in the logging output to save the file.
+
+        <h3>File versioning</h3>
+        If the filename contains a version number, the process will bump the
+        file to the next version after publishing.
+
+        The <code>version</code> field of the resulting <b>Publish</b> in
+        Shotgun will also reflect the version number identified in the filename.
+        The basic worklfow recognizes the following version formats by default:
+
+        <ul>
+        <li><code>filename.v###.ext</code></li>
+        <li><code>filename_v###.ext</code></li>
+        <li><code>filename-v###.ext</code></li>
+        </ul>
+
+        If templates are configured, the version will be determined from the
+        "{version}" token.
+
+        After publishing, if a version number is detected in the work file, the
+        work file will automatically be saved to the next incremental version
+        number. For example, <code>filename.v001.ext</code> will be published
+        and copied to <code>filename.v002.ext</code>
+
+        If the next incremental version of the file already exists on disk, the
+        validation step will produce a warning, and a button will be provided in
+        the logging output which will allow saving the session to the next
+        available version number prior to publishing.
+
+        <br><br><i>NOTE: any amount of version number padding is supported. for
+        non-template based workflows.</i>
+
+        <h3>Overwriting an existing publish</h3>
+        In non-template workflows, a file can be published multiple times,
+        however only the most recent publish will be available to other users.
+        Warnings will be provided during validation if there are previous
+        publishes.
+        """ % (loader_url,)
+
+    @property
+    def settings(self):
+        """
+        Dictionary defining the settings that this plugin expects to receive
+        through the settings parameter in the accept, validate, publish and
+        finalize methods.
+
+        A dictionary on the following form::
+
+            {
+                "Settings Name": {
+                    "type": "settings_type",
+                    "default": "default_value",
+                    "description": "One line description of the setting"
+            }
+
+        The type string should be one of the data types that toolkit accepts as
+        part of its environment configuration.
+        """
+        return {
+            "Publish Type": {
+                "type": "shotgun_publish_type",
+                "default": "Maya Scene",
+                "description": "SG publish type to associate publishes with."
+            },
+            "Work file Template": {
```

Comment:
The templates used for maya session publishing are defined here. Currently, publish2 will not return the actual template object from the `settings` object despite being configured with the `"template"` type. Should it? Right now `settings["Work file Template"].value` returns the name of the configured template (`"maya_shot_work"`).